File: app/src/main/java/com/benoitquenaudon/tvfoot/red/app/injection/module/SchedulerModule.java
package com.benoitquenaudon.tvfoot.red.app.injection.module;

import dagger.Module;
import dagger.Provides;
import com.benoitquenaudon.tvfoot.red.app.common.schedulers.BaseSchedulerProvider;
import com.benoitquenaudon.tvfoot.red.app.common.schedulers.SchedulerProvider;
import javax.inject.Singleton;

@Module public class SchedulerModule {
  @Provides @Singleton BaseSchedulerProvider provideSchedulerProvider() {
    return new SchedulerProvider();
  }
}

File: app/src/main/java/com/benoitquenaudon/tvfoot/red/app/injection/module/SchedulerModule.kt
package com.benoitquenaudon.tvfoot.red.app.injection.module

import com.benoitquenaudon.tvfoot.red.app.common.schedulers.BaseSchedulerProvider
import com.benoitquenaudon.tvfoot.red.app.common.schedulers.SchedulerProvider
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module object SchedulerModule {

  @JvmStatic @Provides @Singleton fun provideSchedulerProvider(): BaseSchedulerProvider {
    return SchedulerProvider()
  }
}
